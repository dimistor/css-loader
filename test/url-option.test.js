import webpack from './helpers/compiler';
import evaluated from './helpers/evaluated';

describe('url', () => {
  test('true', async () => {
    const stats = await webpack('url/url.css');
    const { modules } = stats.toJson();
    const [, , , , , , , , , , , , , module] = modules;

    expect(module.source).toMatchSnapshot('module');
    expect(evaluated(module.source, modules)).toMatchSnapshot(
      'module (evaluated)'
    );
    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });

  test('false', async () => {
    const config = {
      loader: {
        options: {
          url: false,
        },
      },
    };
    const stats = await webpack('url/url.css', config);
    const { modules } = stats.toJson();
    const [, module] = modules;

    expect(module.source).toMatchSnapshot('module');
    expect(evaluated(module.source, modules)).toMatchSnapshot(
      'module (evaluated)'
    );
    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });
});
