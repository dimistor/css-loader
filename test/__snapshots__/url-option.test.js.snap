// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`url false: errors 1`] = `Array []`;

exports[`url false: module (evaluated) 1`] = `
Array [
  Array [
    1,
    "

.class {
  background: url(\\"img.png\\");
}

.class-hash {
  background: url(\\"img.png#hash\\");
}

.class {
  background: url(
    \\"img.png\\"
  );
}

.class {
  background: green url(\\"img.png\\");
}

.class {
  background: url(\\"img.png\\") xyz;
}

.class {
  background: green url(\\"img.png\\") xyz;
}

.class {
  background: url( \\"img.png\\" );
}

.class {
  background: url('img.png');
}

.class {
  background: url(img.png);
}

.class {
  background: url(
    img.png
  );
}

.class {
  background: url( img.png );
}

.class {
  background: url(~package/img.png) url(other-img.png);
}

.class {
  background: green url(~package/img.png) url(other-img.png);
}

.class {
  background: url(~package/img.png) url(other-img.png) xyz;
}

.class {
  background: green url( ~package/img.png ) url(other-img.png) xyz;
}

.class {
  background: url( \\"img img.png\\" );
}

.class {
  background: url( 'img img.png' );
}

.class {
  background: url(/img.png);
}

.class {
  background: url(data:image/png;base64,AAA) url(http://example.com/image.jpg) url(//example.com/image.png);
}

.class {
  background: url(data:image/png;base64,AAA);
}

.class {
  background: url(\\"data:image/svg+xml;charset=utf-8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 42 26' fill='%23007aff'><rect width='4' height='4'/><rect x='8' y='1' width='34' height='2'/><rect y='11' width='4' height='4'/><rect x='8' y='12' width='34' height='2'/><rect y='22' width='4' height='4'/><rect x='8' y='23' width='34' height='2'/></svg>\\");
}

.class {
  background: url(\\"data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%27http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%27%20viewBox%3D%270%200%2042%2026%27%20fill%3D%27%2523007aff%27%3E%3Crect%20width%3D%274%27%20height%3D%274%27%2F%3E%3Crect%20x%3D%278%27%20y%3D%271%27%20width%3D%2734%27%20height%3D%272%27%2F%3E%3Crect%20y%3D%2711%27%20width%3D%274%27%20height%3D%274%27%2F%3E%3Crect%20x%3D%278%27%20y%3D%2712%27%20width%3D%2734%27%20height%3D%272%27%2F%3E%3Crect%20y%3D%2722%27%20width%3D%274%27%20height%3D%274%27%2F%3E%3Crect%20x%3D%278%27%20y%3D%2723%27%20width%3D%2734%27%20height%3D%272%27%2F%3E%3C%2Fsvg%3E\\")
}

.class {
  filter: url('data:image/svg+xml;charset=utf-8,<svg xmlns=\\"http://www.w3.org/2000/svg\\"><filter id=\\"filter\\"><feGaussianBlur in=\\"SourceAlpha\\" stdDeviation=\\"0\\" /><feOffset dx=\\"1\\" dy=\\"2\\" result=\\"offsetblur\\" /><feFlood flood-color=\\"rgba(255,255,255,1)\\" /><feComposite in2=\\"offsetblur\\" operator=\\"in\\" /><feMerge><feMergeNode /><feMergeNode in=\\"SourceGraphic\\" /></feMerge></filter></svg>#filter');
}

.class {
  filter: url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%5C%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%5C%22%3E%3Cfilter%20id%3D%5C%22filter%5C%22%3E%3CfeGaussianBlur%20in%3D%5C%22SourceAlpha%5C%22%20stdDeviation%3D%5C%220%5C%22%20%2F%3E%3CfeOffset%20dx%3D%5C%221%5C%22%20dy%3D%5C%222%5C%22%20result%3D%5C%22offsetblur%5C%22%20%2F%3E%3CfeFlood%20flood-color%3D%5C%22rgba(255%2C255%2C255%2C1)%5C%22%20%2F%3E%3CfeComposite%20in2%3D%5C%22offsetblur%5C%22%20operator%3D%5C%22in%5C%22%20%2F%3E%3CfeMerge%3E%3CfeMergeNode%20%2F%3E%3CfeMergeNode%20in%3D%5C%22SourceGraphic%5C%22%20%2F%3E%3C%2FfeMerge%3E%3C%2Ffilter%3E%3C%2Fsvg%3E%23filter');
}

.class {
  filter: url(#highlight);
}

.class {
  filter: url('#line-marker');
}

.class {
  filter: url(\\"#line-marker\\");
}

@font-face {
  src: url(font.woff) format('woff'),
    url('font.woff2') format('woff2')
    url(\\"font.eot\\") format('eot'),
    url(~package/font.ttf) format('truetype');
}

@font-face {
  src: url('font.eot?#iefix') format('embedded-opentype');
}

@font-face {
  src: url('font.svg#svgFontName') format('svg')
}

@media (min-width: 500px) {
  .class {
    background: url(img.png);
  }
}

.class {
  content: \\"do not use url(path)\\";
}

.class {
  content: 'do not \\"use\\" url(path)';
}

@keyframes anim {
  background: url('img.png');
}

.class {
  background: -webkit-image-set(url('img.png') 1x, url('img2x.png') 2x);
}

.class {
  background: url();
}

.class {
  background: url('');
}

.class {
  background: url('   ');
}

.class {
  background: url(
    ' '
  );
}

.class {
  background: url(https://raw.githubusercontent.com/webpack/media/master/logo/icon.png);
}

.class {
  background: url(//raw.githubusercontent.com/webpack/media/master/logo/icon.png);
}

.foo {
  color: red;
}

.imported-css {
  background: url(./url-imported.css);
}
",
    "",
  ],
]
`;

exports[`url false: module 1`] = `
"// CSS runtime
module.exports = exports = require(\\"../../../src/runtime/api.js\\")(false);

// CSS module
exports.push([module.id, \\"\\\\n\\\\n.class {\\\\n  background: url(\\\\\\"img.png\\\\\\");\\\\n}\\\\n\\\\n.class-hash {\\\\n  background: url(\\\\\\"img.png#hash\\\\\\");\\\\n}\\\\n\\\\n.class {\\\\n  background: url(\\\\n    \\\\\\"img.png\\\\\\"\\\\n  );\\\\n}\\\\n\\\\n.class {\\\\n  background: green url(\\\\\\"img.png\\\\\\");\\\\n}\\\\n\\\\n.class {\\\\n  background: url(\\\\\\"img.png\\\\\\") xyz;\\\\n}\\\\n\\\\n.class {\\\\n  background: green url(\\\\\\"img.png\\\\\\") xyz;\\\\n}\\\\n\\\\n.class {\\\\n  background: url( \\\\\\"img.png\\\\\\" );\\\\n}\\\\n\\\\n.class {\\\\n  background: url('img.png');\\\\n}\\\\n\\\\n.class {\\\\n  background: url(img.png);\\\\n}\\\\n\\\\n.class {\\\\n  background: url(\\\\n    img.png\\\\n  );\\\\n}\\\\n\\\\n.class {\\\\n  background: url( img.png );\\\\n}\\\\n\\\\n.class {\\\\n  background: url(~package/img.png) url(other-img.png);\\\\n}\\\\n\\\\n.class {\\\\n  background: green url(~package/img.png) url(other-img.png);\\\\n}\\\\n\\\\n.class {\\\\n  background: url(~package/img.png) url(other-img.png) xyz;\\\\n}\\\\n\\\\n.class {\\\\n  background: green url( ~package/img.png ) url(other-img.png) xyz;\\\\n}\\\\n\\\\n.class {\\\\n  background: url( \\\\\\"img img.png\\\\\\" );\\\\n}\\\\n\\\\n.class {\\\\n  background: url( 'img img.png' );\\\\n}\\\\n\\\\n.class {\\\\n  background: url(/img.png);\\\\n}\\\\n\\\\n.class {\\\\n  background: url(data:image/png;base64,AAA) url(http://example.com/image.jpg) url(//example.com/image.png);\\\\n}\\\\n\\\\n.class {\\\\n  background: url(data:image/png;base64,AAA);\\\\n}\\\\n\\\\n.class {\\\\n  background: url(\\\\\\"data:image/svg+xml;charset=utf-8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 42 26' fill='%23007aff'><rect width='4' height='4'/><rect x='8' y='1' width='34' height='2'/><rect y='11' width='4' height='4'/><rect x='8' y='12' width='34' height='2'/><rect y='22' width='4' height='4'/><rect x='8' y='23' width='34' height='2'/></svg>\\\\\\");\\\\n}\\\\n\\\\n.class {\\\\n  background: url(\\\\\\"data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%27http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%27%20viewBox%3D%270%200%2042%2026%27%20fill%3D%27%2523007aff%27%3E%3Crect%20width%3D%274%27%20height%3D%274%27%2F%3E%3Crect%20x%3D%278%27%20y%3D%271%27%20width%3D%2734%27%20height%3D%272%27%2F%3E%3Crect%20y%3D%2711%27%20width%3D%274%27%20height%3D%274%27%2F%3E%3Crect%20x%3D%278%27%20y%3D%2712%27%20width%3D%2734%27%20height%3D%272%27%2F%3E%3Crect%20y%3D%2722%27%20width%3D%274%27%20height%3D%274%27%2F%3E%3Crect%20x%3D%278%27%20y%3D%2723%27%20width%3D%2734%27%20height%3D%272%27%2F%3E%3C%2Fsvg%3E\\\\\\")\\\\n}\\\\n\\\\n.class {\\\\n  filter: url('data:image/svg+xml;charset=utf-8,<svg xmlns=\\\\\\"http://www.w3.org/2000/svg\\\\\\"><filter id=\\\\\\"filter\\\\\\"><feGaussianBlur in=\\\\\\"SourceAlpha\\\\\\" stdDeviation=\\\\\\"0\\\\\\" /><feOffset dx=\\\\\\"1\\\\\\" dy=\\\\\\"2\\\\\\" result=\\\\\\"offsetblur\\\\\\" /><feFlood flood-color=\\\\\\"rgba(255,255,255,1)\\\\\\" /><feComposite in2=\\\\\\"offsetblur\\\\\\" operator=\\\\\\"in\\\\\\" /><feMerge><feMergeNode /><feMergeNode in=\\\\\\"SourceGraphic\\\\\\" /></feMerge></filter></svg>#filter');\\\\n}\\\\n\\\\n.class {\\\\n  filter: url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%5C%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%5C%22%3E%3Cfilter%20id%3D%5C%22filter%5C%22%3E%3CfeGaussianBlur%20in%3D%5C%22SourceAlpha%5C%22%20stdDeviation%3D%5C%220%5C%22%20%2F%3E%3CfeOffset%20dx%3D%5C%221%5C%22%20dy%3D%5C%222%5C%22%20result%3D%5C%22offsetblur%5C%22%20%2F%3E%3CfeFlood%20flood-color%3D%5C%22rgba(255%2C255%2C255%2C1)%5C%22%20%2F%3E%3CfeComposite%20in2%3D%5C%22offsetblur%5C%22%20operator%3D%5C%22in%5C%22%20%2F%3E%3CfeMerge%3E%3CfeMergeNode%20%2F%3E%3CfeMergeNode%20in%3D%5C%22SourceGraphic%5C%22%20%2F%3E%3C%2FfeMerge%3E%3C%2Ffilter%3E%3C%2Fsvg%3E%23filter');\\\\n}\\\\n\\\\n.class {\\\\n  filter: url(#highlight);\\\\n}\\\\n\\\\n.class {\\\\n  filter: url('#line-marker');\\\\n}\\\\n\\\\n.class {\\\\n  filter: url(\\\\\\"#line-marker\\\\\\");\\\\n}\\\\n\\\\n@font-face {\\\\n  src: url(font.woff) format('woff'),\\\\n    url('font.woff2') format('woff2')\\\\n    url(\\\\\\"font.eot\\\\\\") format('eot'),\\\\n    url(~package/font.ttf) format('truetype');\\\\n}\\\\n\\\\n@font-face {\\\\n  src: url('font.eot?#iefix') format('embedded-opentype');\\\\n}\\\\n\\\\n@font-face {\\\\n  src: url('font.svg#svgFontName') format('svg')\\\\n}\\\\n\\\\n@media (min-width: 500px) {\\\\n  .class {\\\\n    background: url(img.png);\\\\n  }\\\\n}\\\\n\\\\n.class {\\\\n  content: \\\\\\"do not use url(path)\\\\\\";\\\\n}\\\\n\\\\n.class {\\\\n  content: 'do not \\\\\\"use\\\\\\" url(path)';\\\\n}\\\\n\\\\n@keyframes anim {\\\\n  background: url('img.png');\\\\n}\\\\n\\\\n.class {\\\\n  background: -webkit-image-set(url('img.png') 1x, url('img2x.png') 2x);\\\\n}\\\\n\\\\n.class {\\\\n  background: url();\\\\n}\\\\n\\\\n.class {\\\\n  background: url('');\\\\n}\\\\n\\\\n.class {\\\\n  background: url('   ');\\\\n}\\\\n\\\\n.class {\\\\n  background: url(\\\\n    ' '\\\\n  );\\\\n}\\\\n\\\\n.class {\\\\n  background: url(https://raw.githubusercontent.com/webpack/media/master/logo/icon.png);\\\\n}\\\\n\\\\n.class {\\\\n  background: url(//raw.githubusercontent.com/webpack/media/master/logo/icon.png);\\\\n}\\\\n\\\\n.foo {\\\\n  color: red;\\\\n}\\\\n\\\\n.imported-css {\\\\n  background: url(./url-imported.css);\\\\n}\\\\n\\", \\"\\"]);
"
`;

exports[`url false: warnings 1`] = `Array []`;

exports[`url true: errors 1`] = `Array []`;

exports[`url true: module (evaluated) 1`] = `
Array [
  Array [
    1,
    ".class {
  background: url(/webpack/public/path/img.png);
}

.class-hash {
  background: url(/webpack/public/path/img.png#hash);
}

.class {
  background: url(
    /webpack/public/path/img.png
  );
}

.class {
  background: green url(/webpack/public/path/img.png);
}

.class {
  background: url(/webpack/public/path/img.png) xyz;
}

.class {
  background: green url(/webpack/public/path/img.png) xyz;
}

.class {
  background: url( /webpack/public/path/img.png );
}

.class {
  background: url(/webpack/public/path/img.png);
}

.class {
  background: url(/webpack/public/path/img.png);
}

.class {
  background: url(
    /webpack/public/path/img.png
  );
}

.class {
  background: url( /webpack/public/path/img.png );
}

.class {
  background: url(/webpack/public/path/img.png) url(/webpack/public/path/other-img.png);
}

.class {
  background: green url(/webpack/public/path/img.png) url(/webpack/public/path/other-img.png);
}

.class {
  background: url(/webpack/public/path/img.png) url(/webpack/public/path/other-img.png) xyz;
}

.class {
  background: green url( /webpack/public/path/img.png ) url(/webpack/public/path/other-img.png) xyz;
}

.class {
  background: url( \\"/webpack/public/path/img img.png\\" );
}

.class {
  background: url( \\"/webpack/public/path/img img.png\\" );
}

.class {
  background: url(/img.png);
}

.class {
  background: url(data:image/png;base64,AAA) url(http://example.com/image.jpg) url(//example.com/image.png);
}

.class {
  background: url(data:image/png;base64,AAA);
}

.class {
  background: url(\\"data:image/svg+xml;charset=utf-8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 42 26' fill='%23007aff'><rect width='4' height='4'/><rect x='8' y='1' width='34' height='2'/><rect y='11' width='4' height='4'/><rect x='8' y='12' width='34' height='2'/><rect y='22' width='4' height='4'/><rect x='8' y='23' width='34' height='2'/></svg>\\");
}

.class {
  background: url(\\"data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%27http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%27%20viewBox%3D%270%200%2042%2026%27%20fill%3D%27%2523007aff%27%3E%3Crect%20width%3D%274%27%20height%3D%274%27%2F%3E%3Crect%20x%3D%278%27%20y%3D%271%27%20width%3D%2734%27%20height%3D%272%27%2F%3E%3Crect%20y%3D%2711%27%20width%3D%274%27%20height%3D%274%27%2F%3E%3Crect%20x%3D%278%27%20y%3D%2712%27%20width%3D%2734%27%20height%3D%272%27%2F%3E%3Crect%20y%3D%2722%27%20width%3D%274%27%20height%3D%274%27%2F%3E%3Crect%20x%3D%278%27%20y%3D%2723%27%20width%3D%2734%27%20height%3D%272%27%2F%3E%3C%2Fsvg%3E\\")
}

.class {
  filter: url('data:image/svg+xml;charset=utf-8,<svg xmlns=\\"http://www.w3.org/2000/svg\\"><filter id=\\"filter\\"><feGaussianBlur in=\\"SourceAlpha\\" stdDeviation=\\"0\\" /><feOffset dx=\\"1\\" dy=\\"2\\" result=\\"offsetblur\\" /><feFlood flood-color=\\"rgba(255,255,255,1)\\" /><feComposite in2=\\"offsetblur\\" operator=\\"in\\" /><feMerge><feMergeNode /><feMergeNode in=\\"SourceGraphic\\" /></feMerge></filter></svg>#filter');
}

.class {
  filter: url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%5C%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%5C%22%3E%3Cfilter%20id%3D%5C%22filter%5C%22%3E%3CfeGaussianBlur%20in%3D%5C%22SourceAlpha%5C%22%20stdDeviation%3D%5C%220%5C%22%20%2F%3E%3CfeOffset%20dx%3D%5C%221%5C%22%20dy%3D%5C%222%5C%22%20result%3D%5C%22offsetblur%5C%22%20%2F%3E%3CfeFlood%20flood-color%3D%5C%22rgba(255%2C255%2C255%2C1)%5C%22%20%2F%3E%3CfeComposite%20in2%3D%5C%22offsetblur%5C%22%20operator%3D%5C%22in%5C%22%20%2F%3E%3CfeMerge%3E%3CfeMergeNode%20%2F%3E%3CfeMergeNode%20in%3D%5C%22SourceGraphic%5C%22%20%2F%3E%3C%2FfeMerge%3E%3C%2Ffilter%3E%3C%2Fsvg%3E%23filter');
}

.class {
  filter: url(#highlight);
}

.class {
  filter: url('#line-marker');
}

.class {
  filter: url(\\"#line-marker\\");
}

@font-face {
  src: url(/webpack/public/path/font.woff) format('woff'),
    url(/webpack/public/path/font.woff2) format('woff2')
    url(/webpack/public/path/font.eot) format('eot'),
    url(/webpack/public/path/font.ttf) format('truetype');
}

@font-face {
  src: url(/webpack/public/path/font.eot?#iefix) format('embedded-opentype');
}

@font-face {
  src: url(/webpack/public/path/font.svg#svgFontName) format('svg')
}

@media (min-width: 500px) {
  .class {
    background: url(/webpack/public/path/img.png);
  }
}

.class {
  content: \\"do not use url(path)\\";
}

.class {
  content: 'do not \\"use\\" url(path)';
}

@keyframes anim {
  background: url(/webpack/public/path/img.png);
}

.class {
  background: -webkit-image-set(url(/webpack/public/path/img.png) 1x, url(/webpack/public/path/img2x.png) 2x);
}

.class {
  background: url();
}

.class {
  background: url('');
}

.class {
  background: url('   ');
}

.class {
  background: url(
    ' '
  );
}

.class {
  background: url(https://raw.githubusercontent.com/webpack/media/master/logo/icon.png);
}

.class {
  background: url(//raw.githubusercontent.com/webpack/media/master/logo/icon.png);
}

.foo {
  color: red;
}

.imported-css {
  background: url(14,,);
}
",
    "",
  ],
]
`;

exports[`url true: module 1`] = `
"// CSS runtime escape
var escape = require(\\"../../../src/runtime/escape.js\\");

// CSS runtime
module.exports = exports = require(\\"../../../src/runtime/api.js\\")(false);

// CSS imports
var __url__0__ = escape(require(\\"./img.png\\"));
var __url__1__ = escape(require(\\"./img.png\\"));
var __url__2__ = escape(require(\\"package/img.png\\"));
var __url__3__ = escape(require(\\"./other-img.png\\"));
var __url__4__ = escape(require(\\"./img img.png\\"));
var __url__5__ = escape(require(\\"./font.woff\\"));
var __url__6__ = escape(require(\\"./font.woff2\\"));
var __url__7__ = escape(require(\\"./font.eot\\"));
var __url__8__ = escape(require(\\"package/font.ttf\\"));
var __url__9__ = escape(require(\\"./font.eot\\"));
var __url__10__ = escape(require(\\"./font.svg\\"));
var __url__11__ = escape(require(\\"./img2x.png\\"));
var __url__12__ = escape(require(\\"./url-imported.css\\"));

// CSS module
exports.push([module.id, \\".class {\\\\n  background: url(\\" + __url__0__ + \\");\\\\n}\\\\n\\\\n.class-hash {\\\\n  background: url(\\" + __url__1__ + \\"#hash);\\\\n}\\\\n\\\\n.class {\\\\n  background: url(\\\\n    \\" + __url__0__ + \\"\\\\n  );\\\\n}\\\\n\\\\n.class {\\\\n  background: green url(\\" + __url__0__ + \\");\\\\n}\\\\n\\\\n.class {\\\\n  background: url(\\" + __url__0__ + \\") xyz;\\\\n}\\\\n\\\\n.class {\\\\n  background: green url(\\" + __url__0__ + \\") xyz;\\\\n}\\\\n\\\\n.class {\\\\n  background: url( \\" + __url__0__ + \\" );\\\\n}\\\\n\\\\n.class {\\\\n  background: url(\\" + __url__0__ + \\");\\\\n}\\\\n\\\\n.class {\\\\n  background: url(\\" + __url__0__ + \\");\\\\n}\\\\n\\\\n.class {\\\\n  background: url(\\\\n    \\" + __url__0__ + \\"\\\\n  );\\\\n}\\\\n\\\\n.class {\\\\n  background: url( \\" + __url__0__ + \\" );\\\\n}\\\\n\\\\n.class {\\\\n  background: url(\\" + __url__2__ + \\") url(\\" + __url__3__ + \\");\\\\n}\\\\n\\\\n.class {\\\\n  background: green url(\\" + __url__2__ + \\") url(\\" + __url__3__ + \\");\\\\n}\\\\n\\\\n.class {\\\\n  background: url(\\" + __url__2__ + \\") url(\\" + __url__3__ + \\") xyz;\\\\n}\\\\n\\\\n.class {\\\\n  background: green url( \\" + __url__2__ + \\" ) url(\\" + __url__3__ + \\") xyz;\\\\n}\\\\n\\\\n.class {\\\\n  background: url( \\" + __url__4__ + \\" );\\\\n}\\\\n\\\\n.class {\\\\n  background: url( \\" + __url__4__ + \\" );\\\\n}\\\\n\\\\n.class {\\\\n  background: url(/img.png);\\\\n}\\\\n\\\\n.class {\\\\n  background: url(data:image/png;base64,AAA) url(http://example.com/image.jpg) url(//example.com/image.png);\\\\n}\\\\n\\\\n.class {\\\\n  background: url(data:image/png;base64,AAA);\\\\n}\\\\n\\\\n.class {\\\\n  background: url(\\\\\\"data:image/svg+xml;charset=utf-8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 42 26' fill='%23007aff'><rect width='4' height='4'/><rect x='8' y='1' width='34' height='2'/><rect y='11' width='4' height='4'/><rect x='8' y='12' width='34' height='2'/><rect y='22' width='4' height='4'/><rect x='8' y='23' width='34' height='2'/></svg>\\\\\\");\\\\n}\\\\n\\\\n.class {\\\\n  background: url(\\\\\\"data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%27http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%27%20viewBox%3D%270%200%2042%2026%27%20fill%3D%27%2523007aff%27%3E%3Crect%20width%3D%274%27%20height%3D%274%27%2F%3E%3Crect%20x%3D%278%27%20y%3D%271%27%20width%3D%2734%27%20height%3D%272%27%2F%3E%3Crect%20y%3D%2711%27%20width%3D%274%27%20height%3D%274%27%2F%3E%3Crect%20x%3D%278%27%20y%3D%2712%27%20width%3D%2734%27%20height%3D%272%27%2F%3E%3Crect%20y%3D%2722%27%20width%3D%274%27%20height%3D%274%27%2F%3E%3Crect%20x%3D%278%27%20y%3D%2723%27%20width%3D%2734%27%20height%3D%272%27%2F%3E%3C%2Fsvg%3E\\\\\\")\\\\n}\\\\n\\\\n.class {\\\\n  filter: url('data:image/svg+xml;charset=utf-8,<svg xmlns=\\\\\\"http://www.w3.org/2000/svg\\\\\\"><filter id=\\\\\\"filter\\\\\\"><feGaussianBlur in=\\\\\\"SourceAlpha\\\\\\" stdDeviation=\\\\\\"0\\\\\\" /><feOffset dx=\\\\\\"1\\\\\\" dy=\\\\\\"2\\\\\\" result=\\\\\\"offsetblur\\\\\\" /><feFlood flood-color=\\\\\\"rgba(255,255,255,1)\\\\\\" /><feComposite in2=\\\\\\"offsetblur\\\\\\" operator=\\\\\\"in\\\\\\" /><feMerge><feMergeNode /><feMergeNode in=\\\\\\"SourceGraphic\\\\\\" /></feMerge></filter></svg>#filter');\\\\n}\\\\n\\\\n.class {\\\\n  filter: url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%5C%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%5C%22%3E%3Cfilter%20id%3D%5C%22filter%5C%22%3E%3CfeGaussianBlur%20in%3D%5C%22SourceAlpha%5C%22%20stdDeviation%3D%5C%220%5C%22%20%2F%3E%3CfeOffset%20dx%3D%5C%221%5C%22%20dy%3D%5C%222%5C%22%20result%3D%5C%22offsetblur%5C%22%20%2F%3E%3CfeFlood%20flood-color%3D%5C%22rgba(255%2C255%2C255%2C1)%5C%22%20%2F%3E%3CfeComposite%20in2%3D%5C%22offsetblur%5C%22%20operator%3D%5C%22in%5C%22%20%2F%3E%3CfeMerge%3E%3CfeMergeNode%20%2F%3E%3CfeMergeNode%20in%3D%5C%22SourceGraphic%5C%22%20%2F%3E%3C%2FfeMerge%3E%3C%2Ffilter%3E%3C%2Fsvg%3E%23filter');\\\\n}\\\\n\\\\n.class {\\\\n  filter: url(#highlight);\\\\n}\\\\n\\\\n.class {\\\\n  filter: url('#line-marker');\\\\n}\\\\n\\\\n.class {\\\\n  filter: url(\\\\\\"#line-marker\\\\\\");\\\\n}\\\\n\\\\n@font-face {\\\\n  src: url(\\" + __url__5__ + \\") format('woff'),\\\\n    url(\\" + __url__6__ + \\") format('woff2')\\\\n    url(\\" + __url__7__ + \\") format('eot'),\\\\n    url(\\" + __url__8__ + \\") format('truetype');\\\\n}\\\\n\\\\n@font-face {\\\\n  src: url(\\" + __url__9__ + \\"?#iefix) format('embedded-opentype');\\\\n}\\\\n\\\\n@font-face {\\\\n  src: url(\\" + __url__10__ + \\"#svgFontName) format('svg')\\\\n}\\\\n\\\\n@media (min-width: 500px) {\\\\n  .class {\\\\n    background: url(\\" + __url__0__ + \\");\\\\n  }\\\\n}\\\\n\\\\n.class {\\\\n  content: \\\\\\"do not use url(path)\\\\\\";\\\\n}\\\\n\\\\n.class {\\\\n  content: 'do not \\\\\\"use\\\\\\" url(path)';\\\\n}\\\\n\\\\n@keyframes anim {\\\\n  background: url(\\" + __url__0__ + \\");\\\\n}\\\\n\\\\n.class {\\\\n  background: -webkit-image-set(url(\\" + __url__0__ + \\") 1x, url(\\" + __url__11__ + \\") 2x);\\\\n}\\\\n\\\\n.class {\\\\n  background: url();\\\\n}\\\\n\\\\n.class {\\\\n  background: url('');\\\\n}\\\\n\\\\n.class {\\\\n  background: url('   ');\\\\n}\\\\n\\\\n.class {\\\\n  background: url(\\\\n    ' '\\\\n  );\\\\n}\\\\n\\\\n.class {\\\\n  background: url(https://raw.githubusercontent.com/webpack/media/master/logo/icon.png);\\\\n}\\\\n\\\\n.class {\\\\n  background: url(//raw.githubusercontent.com/webpack/media/master/logo/icon.png);\\\\n}\\\\n\\\\n.foo {\\\\n  color: red;\\\\n}\\\\n\\\\n.imported-css {\\\\n  background: url(\\" + __url__12__ + \\");\\\\n}\\\\n\\", \\"\\"]);
"
`;

exports[`url true: warnings 1`] = `Array []`;
